src/integrations/pathManager.ts

```diff
@@ -65,9 +65,57 @@ export class RcFilePathManager implements PathManager {
     await this.manageFish(false);
   }
 
+  /**
+   * bash requires some special handling. This is because the files it reads
+   * on startup differ depending on whether it is a login shell or a
+   * non-login shell. We must cover both cases.
+   */
   protected async managePosix(desiredPresent: boolean): Promise<void> {
     const pathLine = `export PATH="${ paths.integration }:$PATH"`;
+    // Note: order is important here. Only the first one that is present is modified.
+    const bashLoginShellFiles = [
+      '.bash_profile',
+      '.bash_login',
+      '.profile',
+    ];
+
+    // Handle files that pertain to bash login shells
+    if (desiredPresent) {
+      let linesAdded = false;
+
+      // Write the first file that exists, if any
+      for (const fileName of bashLoginShellFiles) {
+        const filePath = path.join(os.homedir(), fileName);
+
+        try {
+          await fs.promises.stat(filePath);
+        } catch (error: any) {
+          if (error.code === 'ENOENT') {
+            continue;
+          }
+          throw error;
+        }
+        await manageLinesInFile(filePath, [pathLine], desiredPresent);
+        linesAdded = true;
+        break;
+      }
+
+      // If none of the files exist, write .bash_profile
+      if (!linesAdded) {
+        const filePath = path.join(os.homedir(), '.bash_profile');
+
+        await manageLinesInFile(filePath, [pathLine], desiredPresent);
+      }
+    } else {
+      // Ensure lines are not present in any of the files
+      await Promise.all(bashLoginShellFiles.map((fileName) => {
+        const filePath = path.join(os.homedir(), fileName);
+
+        return manageLinesInFile(filePath, [], desiredPresent);
```

Comment:
Amusingly, this means if the user had a blank `.bash_profile` and a non-blank `.profile` before RD was ever involved, and we wanted to remove this, we end up deleting `.bash_profile` and cause their `.profile` to come into effect.

I'm not sure I care…

Reply:
As bad as it sounds, this seems like a user problem. We can't account for every little difference in system setup :disappointed:

Reply:
Right, it's probably fine to ignore that.

Reply:
I agree that it is an unlikely scenario, but isn't it easily fixed by not deleting the file, even if it is empty after removing our modifications?

This may leave you with an empty `.bash_profile` if you never had a `.bash_profile`, `.bash_login`, or `.bashrc`, but having an empty `.bash_profile` seems like a non-issue.

Why do we delete empty files?